var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -512,6 +512,16 @@ def cmake_args(self):
                 options.append("-DGMX_GPU:STRING=CUDA")
             elif "+opencl" in self.spec:
                 options.append("-DGMX_GPU:STRING=OpenCL")
+            elif "+rocm" in self.spec:
```

Comment:
> If we build gromacs+sycl+rocm, it pulls hipsycl~rocm as a dependency, which would not work.

To clarify: you're right that this combination was allowed (although it was not strictly required)

I added a constraint for now to force `gromacs+rocm` to build against `hipsycl+rocm`

> hip pulls a ton of things, including mesa, which is definitely not required for our use; is there a way to limit that? Could we only require a minimal set?

That could work although this all comes from inheriting `ROCmPackage` (which includes standardizations that are helpful in maintaining idioms WRT handling ROCm across packages, e.g. `amdgpu_target`), so I'd prefer to update that to make things like `mesa` optional (I'm guessing this would be useful for many packages that inherit `ROCmPackage`). In my case, I was using an external set of ROCm dependencies, which avoids building many dependencies.

Reply:
> That could work although this all comes from inheriting ROCmPackage (which includes standardizations that are helpful in maintaining idioms WRT handling ROCm across packages, e.g. amdgpu_target), so I'd prefer to update that to make things like mesa optional (I'm guessing this would be useful for many packages that inherit ROCmPackage). In my case, I was using an external set of ROCm dependencies, which avoids building many dependencies.

I see, thanks for explaining! If that's not something that can be easily dealt with here, let's not worry about it :)